ext/flash/01_http.js

```diff
@@ -456,29 +380,111 @@
               }
             }
 
-            if (ws) {
-              const wsRid = await core.opAsync(
-                "op_flash_upgrade_websocket",
-                serverId,
-                i,
-              );
-              ws[_rid] = wsRid;
-              ws[_protocol] = resp.headers.get("sec-websocket-protocol");
-
-              ws[_readyState] = WebSocket.OPEN;
-              const event = new Event("open");
-              ws.dispatchEvent(event);
-
-              ws[_eventLoop]();
-              if (ws[_idleTimeoutDuration]) {
-                ws.addEventListener(
-                  "close",
-                  () => clearTimeout(ws[_idleTimeoutTimeout]),
+            (async () => {
+              if (!ws) {
+                if (hasBody && body[_state] !== "closed") {
+                  // TODO(@littledivy): Optimize by draining in a single op.
+                  try {
+                    await req.arrayBuffer();
+                  } catch { /* pass */ }
+                }
+              }
+
+              if (isStreamingResponseBody === true) {
+                const resourceRid = getReadableStreamRid(respBody);
+                if (resourceRid) {
+                  if (respBody.locked) {
+                    throw new TypeError("ReadableStream is locked.");
```

Comment:
This will be unhandled rejection, I suggest to pipe all errors to `onError` handler